### ISSUE TITLE
Next.js plug-in timeout 

### ISSUE THREAD

**Issue by stefee** [fastify/help#663]
## 💬  Question here 

I find the fastify-nextjs plugin will start to timeout during initialisation during development mode, because the application code needs to compile. I’m wondering if there are strategies we can use to mitigate this beyond increasing the plug-in timeout for the entire Fastify instance globally. 

For example, can we increase the timeout only on that specific register call?

Could the default timeout be increased on the plug-in itself during dev, since this issue will likely affect most use cases when dev mode is used.

**Comment by stefee** [comment]
I’m wondering if it’s necessary to block the entire plug-in startup here? 

https://github.com/fastify/fastify-nextjs/blob/3affd5cd6e654ff36c0ad5a87b7038a2afbad69c/index.js#L21


Could this promise be awaited lazily during request handler instead? 🤔

**Comment by Eomm** [comment]
> Could the default timeout be increased on the plug-in itself during dev, since this issue will likely affect most use cases when dev mode is used.

No, there is no way for a plugin to set the fastify's server config

The only way is to configure your env:

```
fastify({
  pluginTimeout: process.env.FASTIFY_PLUGIN_TIMEOUT
})
```

> Could this promise be awaited lazily during request handler instead? 🤔

PRs are welcome in that sense
